src/api/route-services/work-response.js

```diff
@@ -1,28 +1,17 @@
-const yaml = require('js-yaml');
-const fs = require('fs');
-const path = require('path');
-const Validator = require('swagger-model-validator');
 const AWS = require('aws-sdk');
+const validateRequest = require('../../utils/schema-validator');
 const logger = require('../../utils/logging');
 const { cacheSetResponse } = require('../../utils/cache-request');
 const { handlePagination } = require('../../utils/handlePagination');
 
 class WorkResponseService {
   constructor(io, workResponse) {
-    const specPath = path.resolve(__dirname, '..', '..', 'specs', 'api.yaml');
-    const specObj = yaml.safeLoad(fs.readFileSync(specPath), 'utf8');
-    const validator = new Validator();
-    const res = validator.validate(
-      workResponse, specObj.components.schemas.WorkResponse, specObj.components.schemas,
-    );
-
-
-    if (!res.valid) {
-      throw new Error(res.errors);
-    }
-
-    this.workResponse = workResponse;
-    this.io = io;
+    return (async () => {
+      await validateRequest(workResponse, 'WorkRequest.v1.yaml');
```

Comment:
nice! We have to test this thoroughly though if there are any existing work responses that are not according to the schema format

Reply:
also, shouldn't this be validated with the `WorkResponse` schema instead of the `WorkRequest`?

Reply:
also, what happens if there is a validation error? Do we log it anywhere? We don't want the API to silently fail if there is a validation error, because it will be super hard to debug.

Reply:
For (1), we don't actually validate based on the work response schemas right now, as we only have a limited number of them. You can see this here -- response is just any object, no requirements as to its contents.

![image](https://user-images.githubusercontent.com/16019444/107951116-bb669200-6f8f-11eb-9d89-b09aef7b2dcc.png)

Reply:
For (2), yes, that is a typo. Will fix.

Reply:
For (3), the same exact thing happens as right now when there is a validation error. The error is thrown into the handler for `/workResult`, which does not handle it, so it ends up in the main error handler of the API and logged.